pyxdf/pyxdf.py

```diff
@@ -242,11 +240,17 @@ def load_xdf(filename,
                 # read [NumLengthBytes], [Length]
                 chunklen = _read_varlen_int(f)
             except Exception:
-                if f.tell() < filesize - 1024:
+                # If there's more data available (i.e. a read() succeeds),
+                # find the next boundary chunk
+                if f.read(1):
                     logger.warning('got zero-length chunk, scanning forward to '
                                    'next boundary chunk.')
-                    _scan_forward(f)
-                    continue
+                    # move the stream position one byte back
+                    f.seek(-1, 1)
+                    if _scan_forward(f):
+                        continue
+                    else:
```

Comment:
I guess you could get rid of this `else: break` block because you have the same thing right in the next `else` block. I.e. I think you can remove the first `else` block completely (the one belonging to the `if _scan_forward(f)`), and also remove the second `else` and dedent the two subsequent lines.

Reply:
Right, I've changed it.